src/metabase/models/user.clj

```diff
@@ -129,14 +133,15 @@
 
 (defn invite-user!
   "Convenience function for inviting a new `User` and sending out the welcome email."
-  [first-name last-name email-address password invitor]
+  [first-name last-name email-address password login-attributes invitor]
```

Comment:
personally I think functions get a bit hard to read/use when they have more than 5 ordered args I would personally vote that we switch to keyword args at this point.

Also maybe we can switch this function to use a schema for validation while we're at it instead of old-skool preconditions. (Just a thought)

Reply:
I spent about an hour or so on just what you describe yesterday and wasn't happy with the result. I think I was too ambitious, I'll try it again.